src/components/Missions.js

```diff
@@ -1,41 +1,63 @@
-// IMPORTS
-import { useEffect } from 'react';
+import React, { useEffect } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
-import { getMissions } from '../redux/missionsSlice';
-// import '../styles/missions.css';
+import { Button, Table } from 'react-bootstrap';
+import { getMissions, joinMission, leaveMission } from '../redux/missionsSlice';
+import '../styles/missions.css';
 
-// GET MISSIONS DATA FROM REDUX STORE
 const Missions = () => {
   const dispatch = useDispatch();
   const missions = useSelector((state) => state.missions.missions);
-  //  console.log('Missions:', missions);
 
-  // FETCH MISSIONS DATA ON PAGE LOAD
   useEffect(() => {
     dispatch(getMissions());
   }, [dispatch]);
 
-  // SHOW LOADING, ERROR, OR MISSIONS DATA
+  const handleJoinMission = (id, joined) => {
+    if (joined) {
+      dispatch(leaveMission(id));
+    } else {
+      dispatch(joinMission(id));
+    }
+  };
+
```

Comment:
- [ ] Good job implementing the space traveler app 👏. It looks good however I believe you can do better. Here is the description of the thing I suggest you to improve. Whenever the Mission component is rendered , the `state` of all mission is reset 😄 . Which cause the Missions the used has joined to be reset. To fix that , you should dispatch the  `getMissions` in the `App.js` because it is called once 👍 . Kindly update that


```jsx
...
import { useEffect } from 'react'; 
import { getMissions } from './redux/missionsSlice';
...

function App() {
  const dispatch = useDispatch();
    useEffect(() => {
    dispatch(getMissions());
  }, [dispatch]);
...
}
```